man/sysupdate.d.xml

```diff
@@ -0,0 +1,872 @@
+
+        <para>Like many of the settings in these configuration files this setting supports specifier
+        expansion. It's particularly useful to set this setting to one of <literal>%A</literal>,
+        <literal>%B</literal> or <literal>%w</literal> to automatically refer to the current OS version of
+        the running system. See below for details on supported specifiers.</para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>Verify=</varname></term>
+
+        <listitem><para>Takes a boolean, defaults to yes. Controls whether to cryptographically verify
+        downloaded resources (specifically: validate the GPG signatures for downloaded
+        <filename>SHA256SUMS</filename> manifest files, via their detached signature files
+        <filename>SHA256SUMS.gpg</filename>).</para>
+
+        <para>This option is essential to provide integrity guarantees for downloaded resources and thus
+        should be left enabled, outside of test environments.</para>
+
+        <para>Note that the downloaded payload files are unconditionally checked against the SHA256 hashes
+        listed in the manifest. This option only controls whether the signatures of these manifests are
+        verified.</para>
+
+        <para>This option only has an effect if the source resource type is selected as
+        <constant>http-file</constant> or <constant>http-tar</constant>, as integrity and authentication
+        checking is only available for transfers from remote sources.</para></listitem>
+      </varlistentry>
+
+    </variablelist>
+  </refsect1>
+
+
+  <refsect1>
+    <title>[Source] Section Options</title>
+
+    <para>This section defines properties of the transfer source:</para>
+
+    <variablelist>
+      <varlistentry>
+        <term><varname>Type=</varname></term>
+
+        <listitem><para>Specifies the resource type of the source for the transfer. Takes one of
+        <constant>http-file</constant>, <constant>http-tar</constant>, <constant>tar</constant>,
+        <constant>regular-file</constant>, <constant>directory</constant> or
+        <constant>subvolume</constant>. For details about the resource types, see above. This option is
+        mandatory.</para>
+
+        <para>Note that only some combinations of source and target resource types are supported, see
+        above.</para></listitem>
+      </varlistentry>
+    </variablelist>
+
+    <variablelist>
+      <varlistentry>
+        <term><varname>Path=</varname></term>
+
+        <listitem><para>Specifies where to find source versions of this resource.</para>
+
+        <para>If the source type is selected as <constant>http-file</constant> or
+        <constant>http-tar</constant> this must be a HTTP/HTTPS URI. The URI is suffixed with
+        <filename>/SHA256SUMS</filename> to acquire the manifest file, with
+        <filename>/SHA256SUMS.gpg</filename> to acquire the detached signature file for it, and with the file
```

Comment:
That seems to hardcode the favorite hash func of the day. I'm pretty sure it should be configurable. In particular, Fedora uses SHA512 for its dist-git cache.

Reply:
fedora uses basically randomized names for the SHA512SUMS file. it's really bad. You can't easily figure out what the file is you are looking for without some external info.

but yeah, we need to extend this eventually. We inherit this from importd though, it's what it always supported. importd can also do suse style checksum files (which is a bit confused, given they name the file after the images, so they are obviously deterministic but not useful to enumerate images).

But this should not be misunderstood: this is not an attempt to support every format in the world. Instead it's an attempt to do version management with only "standard UNIX tools" and the least amount of metadata we can get away with if you so will, and in a way that happens to be compatible with one major distro that did things the right way (in my opinion: ubuntu). It's an excercise in avoiding to define new metadata formats (which is what everyone else does), by trying to be smart and using an established format with absolutely minimal footprint.

At this point in time SHA256 should be totally fine. Adding support for SHA512 is totally doable, it's a minor addition really, the file format remains the same after all. But for now I don't think we should bother, the simplicity of just having one way to do things for now is extremely attractive to me — as long as we can naturally extend this later for SHA512. Which we can.

Reply:
> fedora uses basically randomized names for the SHA512SUMS file. it's really bad. You can't easily figure out what the file is you are looking for without some external info

I don't grok this: the file always called `sources` and it lists the files by the name of the file, as specified by the packager, but that is supposed just use what upstream uses. So there is some unpredictability in the naming, but that's because of upstream conventions but in general you know what the file is from the name...

I think it's better to take into account an eventual hash renaming… Let's take for example the process in Fedora: there old format had md5 hashes just as a raw number and file name. When transitioning to sha, they added a marker for the hash function. The file name stayed the same. The transition was fairly nice: only tools that look at the `sources` content needed to be adjusted, and the transition is forward-compatible in the sense that you can have a different hash function, or multiple hash functions in the same file.

This isn't terribly imporant… But I think that hardcoding the hash func name in such a visible fashion is an antipattern that we know how to easily avoid.